ui/app/components/http-requests-bar-chart.js

```diff
@@ -0,0 +1,206 @@
+import Component from '@ember/component';
+import d3 from 'd3-selection';
+import d3Scale from 'd3-scale';
+import d3Axis from 'd3-axis';
+import d3TimeFormat from 'd3-time-format';
+import { assign } from '@ember/polyfills';
+import { computed } from '@ember/object';
+import { run } from '@ember/runloop';
+import { task, waitForEvent } from 'ember-concurrency';
+
+/**
+ * @module HttpRequestsBarChart
+ * HttpRequestsBarChart components are used to render a bar chart with the total number of HTTP Requests to a Vault server per month.
+ *
+ * @example
+ * ```js
+ * <HttpRequestsBarChart @counters={{counters}} />
+ * ```
+ *
+ * @param counters=null {Array} - A list of objects containing the total number of HTTP Requests for each month. `counters` should be the response from the `/internal/counters/requests` endpoint which looks like:
+ * COUNTERS = [
+ *    {
+ *       "start_time": "2019-05-01T00:00:00Z",
+ *       "total": 50
+ *     }
+ * ]
+ */
+
+const COUNTERS = [
+  {
+    start_time: '2019-05-01T00:00:00Z',
+    total: 50000,
+  },
+  {
+    start_time: '2019-04-01T00:00:00Z',
+    total: 4500,
+  },
+  {
+    start_time: '2019-03-01T00:00:00Z',
+    total: 550000,
+  },
+];
+
+export default Component.extend({
+  classNames: ['http-requests-bar-chart-container'],
+  counters: COUNTERS,
+  svgContainer: null,
+  barsContainer: null,
+  clipPathRect: null,
+  margin: { top: 24, right: 16, bottom: 24, left: 16 },
+  width: 0,
+  height() {
+    const { margin } = this;
+    return 240 - margin.top - margin.bottom;
+  },
+
+  parsedCounters: computed('counters', function() {
+    // parse the start times so bars and ticks display properly
+    const { counters } = this;
+    return counters.map(counter => {
+      return assign({}, counter, { start_time: d3TimeFormat.isoParse(counter.start_time) });
+    });
+  }),
+
+  yScale: computed('parsedCounters', 'height', function() {
+    const { parsedCounters } = this;
+    const height = this.height();
+    const counterTotals = parsedCounters.map(c => c.total);
+
+    return d3Scale
+      .scaleLinear()
+      .domain([0, Math.max(...counterTotals)])
```

Comment:
d3 has some nice helpers to do this in one pass, but don't think it's necessary to pull them in here https://github.com/d3/d3-array so it'd be `d3.min(parsedCounters, d=> d.total)`

Reply:
i looked into that, but i think we always want the y axis start at 0 and end with the highest number in the list. i didn't see a benefit to using `d3.max` vs. `Math.max` -- is there one i'm not aware of? 🤔

Reply:
Heh sorry missed your reply the first time around - the advantage is just that it's a shortcut, you wouldn't have to compute counterTotals. But you're right in this case it's just one place so the shortcut isn't super helpful.